transcom/mymove
# B 18411 UI shipment address update review screen updates

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A882385)

## Summary

In order to get the full functionality, please run `git merge origin/B-18475-db-backend-updates-shipment-dest-address-update-sit` to merge the backend portion of this feature into the branch.
[B-18475 PR](https://github.com/transcom/mymove/pull/11790)

When the TOO reviews a shipment destination address update request - there's a reason for it. It means that there was a check done in the backend that changed the status of the request in the `shipment_address_updates` table to `REQUESTED` instead of automatically changing it to `APPROVED`.

When a shipment contains destination SIT service items, the above mentioned backend branch does logic checks that populates the table with SIT data. 

This PR does the following:
- takes that SIT data sent to the UI and returns an info banner if the new mileage change for a shipment containing SIT is changed to **greater than** 50 miles

### How to test

1. Access MM as prime
2. Request a shipment destination address update to a ZIP that is > 50 miles of what the current destination address is
3. Log in as the TOO & navigate to that shipment
4. Follow the red prompts until you get to the address update modal that shows the changes
5. You should see an info banner that shows the updated mileage and a general statement to the TOO letting them know what's up

## Screenshots

![Screenshot 2024-01-23 at 10 58 24 AM](https://github.com/transcom/mymove/assets/136510600/20031237-90ad-4d1a-be1b-6d74b5ca22dd)
